File: src/main/java/p00/Problem_0015.java
package p00;

import java.math.BigInteger;

public class Problem_0015 {

	public static BigInteger solve() {
		return Faculty(40).divide(Faculty(20).pow(2));
	}

	private static BigInteger Faculty(int n) {
		BigInteger result = BigInteger.ONE;

		for (int i = 2; i <= n; i++)
			result = result.multiply(BigInteger.valueOf(i));

		return result;
	}

}

File: src/main/kotlin/p00/Problem0015.kt
package p00

import java.math.BigInteger

class Problem0015 {
    fun solve(): BigInteger {
        return faculty(40).divide(faculty(20).pow(2))
    }

    private fun faculty(n: Int): BigInteger {
        var result = BigInteger.ONE
        for (i in 2..n) result = result.multiply(BigInteger.valueOf(i.toLong()))
        return result
    }
}